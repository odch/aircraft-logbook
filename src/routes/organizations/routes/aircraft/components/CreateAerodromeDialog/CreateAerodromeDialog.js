import React from 'react'
import PropTypes from 'prop-types'
import { injectIntl, FormattedMessage } from 'react-intl'
import Button from '@material-ui/core/Button'
import Dialog from '@material-ui/core/Dialog'
import DialogActions from '@material-ui/core/DialogActions'
import DialogContent from '@material-ui/core/DialogContent'
import DialogTitle from '@material-ui/core/DialogTitle'
import TextField from '@material-ui/core/TextField'
import CircularProgress from '@material-ui/core/CircularProgress'
import { withStyles } from '@material-ui/core/styles'
import moment from 'moment-timezone'
import { intl as intlShape } from '../../../../../../shapes'
import Select from '../../../../../../components/Select'

const styles = {
  root: { overflow: 'visible' },
  loadingIndicator: {
    marginRight: 5
  }
}

const TIMEZONE_OPTIONS = moment.tz
  .names()
  .filter(name => name.includes('/'))
  .map(name => ({
    value: name,
    label: name
  }))

class CreateAerodromeDialog extends React.Component {
  handleTextChange = name => e => {
    this.updateData(name, e.target.value)
  }

  handleIdentificationChange = e => {
    this.updateData('identification', e.target.value.toUpperCase())
  }

  handleSelectChange = name => value => {
    this.updateData(name, value)
  }

  updateData = (name, value) => {
    this.props.updateData({
      [name]: value
    })
  }

  handleClose = () => {
    const { onClose, submitting } = this.props

    if (submitting !== true && onClose) {
      onClose()
    }
  }

  handleSubmit = e => {
    const { onSubmit, organizationId, fieldName, data, submitting } = this.props

    e.preventDefault()
    if (submitting !== true && onSubmit) {
      onSubmit(organizationId, fieldName, data)
    }
  }

  msg = id => this.props.intl.formatMessage({ id })

  render() {
    const { submitting, classes, onClose } = this.props
    return (
      <Dialog
        classes={{ paperScrollPaper: classes.root }}
        onClose={this.handleClose}
        data-cy="aerodrome-create-dialog"
        open
      >
        <DialogTitle>
          <FormattedMessage id="organization.aerodrome.create.dialog.title" />
        </DialogTitle>
        <form onSubmit={this.handleSubmit}>
          <DialogContent className={classes.root}>
            {this.renderTextField(
              'identification',
              true,
              true,
              this.handleIdentificationChange
            )}
            {this.renderTextField('name', false, true)}
            {this.renderSelect('timezone', TIMEZONE_OPTIONS, true)}
          </DialogContent>
          <DialogActions>
            <Button onClick={onClose} disabled={submitting}>
              <FormattedMessage id="organization.aerodrome.create.dialog.buttons.cancel" />
            </Button>
            <Button
              type="submit"
              color="primary"
              variant="contained"
              data-cy="create-button"
              disabled={submitting}
            >
              {submitting && (
                <CircularProgress
                  size={16}
                  className={classes.loadingIndicator}
                />
              )}
              <FormattedMessage id="organization.aerodrome.create.dialog.buttons.create" />
            </Button>
          </DialogActions>
        </form>
      </Dialog>
    )
  }

  renderTextField(
    name,
    autoFocus,
    required,
    onChange = this.handleTextChange(name)
  ) {
    return (
      <TextField
        label={this.msg(`organization.aerodrome.create.dialog.${name}`)}
        type="text"
        fullWidth
        required={required}
        value={this.props.data[name]}
        onChange={onChange}
        data-cy={`${name}-field`}
        disabled={this.props.submitting}
        autoFocus={autoFocus}
      />
    )
  }

  renderSelect(name, options, required) {
    return (
      <Select
        label={this.msg(`organization.aerodrome.create.dialog.${name}`)}
        type="text"
        fullWidth
        required={required}
        value={this.props.data[name]}
        onChange={this.handleSelectChange(name)}
        data-cy={`${name}-field`}
        disabled={this.props.submitting}
        options={options}
      />
    )
  }
}

CreateAerodromeDialog.propTypes = {
  organizationId: PropTypes.string.isRequired,
  fieldName: PropTypes.string.isRequired,
  data: PropTypes.shape({
    identification: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    timezone: PropTypes.shape({
      value: PropTypes.string,
      label: PropTypes.string
    })
  }).isRequired,
  submitting: PropTypes.bool,
  onClose: PropTypes.func,
  onSubmit: PropTypes.func,
  updateData: PropTypes.func.isRequired,
  intl: intlShape,
  classes: PropTypes.object.isRequired
}

export default injectIntl(withStyles(styles)(CreateAerodromeDialog))
