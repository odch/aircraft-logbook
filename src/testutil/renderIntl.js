import React from 'react'
import renderer from 'react-test-renderer'
import { createMount } from '@material-ui/core/test-utils'
import { IntlProvider } from 'react-intl'
import pretty from 'pretty'
import messages from '../messages'

const LOCALE = 'de'

const renderInIntlProvider = component => (
  <IntlProvider locale={LOCALE} messages={messages[LOCALE]}>
    {component}
  </IntlProvider>
)

const renderIntl = (component, options) =>
  renderer.create(renderInIntlProvider(component), options)

export const renderIntlUpdate = (renderer, component, options) =>
  renderer.update(renderInIntlProvider(component), options)

export const renderIntlMaterial = (component, returnPrettyHtml = false) => {
  const rendered = createMount()(renderInIntlProvider(component))
  return returnPrettyHtml ? pretty(rendered.html()) : rendered
}

export default renderIntl
