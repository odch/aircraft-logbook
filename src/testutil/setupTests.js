import * as enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import moment from 'moment'
import momentTimezone from 'moment-timezone'
import { init as initFeatureToggles } from '../util/featureToggles'

enzyme.configure({ adapter: new Adapter() })

moment.locale('de')
momentTimezone.locale('de')

initFeatureToggles()
