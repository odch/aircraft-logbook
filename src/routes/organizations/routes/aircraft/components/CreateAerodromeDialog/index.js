import CreateAerodromeDialog from './CreateAerodromeDialog'

export default CreateAerodromeDialog
