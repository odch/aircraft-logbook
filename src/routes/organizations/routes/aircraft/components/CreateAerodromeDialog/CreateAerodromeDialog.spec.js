import React from 'react'
import { renderIntlMaterial } from '../../../../../../testutil/renderIntl'
import CreateAerodromeDialog from './CreateAerodromeDialog'

describe('routes', () => {
  describe('organizations', () => {
    describe('routes', () => {
      describe('aircraft', () => {
        describe('components', () => {
          describe('CreateAerodromeDialog', () => {
            it('renders correctly', () => {
              const renderedValue = renderIntlMaterial(
                <CreateAerodromeDialog
                  organizationId="my_org"
                  data={{ firstname: 'Max', lastname: 'Muster', id: '24354' }}
                  updateData={() => {}}
                  open
                />,
                true
              )
              expect(renderedValue).toMatchSnapshot()
            })

            it('renders readonly if submitting', () => {
              const renderedValue = renderIntlMaterial(
                <CreateAerodromeDialog
                  organizationId="my_org"
                  data={{ firstname: 'Max', lastname: 'Muster', id: '24354' }}
                  updateData={() => {}}
                  submitting
                />,
                true
              )
              expect(renderedValue).toMatchSnapshot()
            })
          })
        })
      })
    })
  })
})
