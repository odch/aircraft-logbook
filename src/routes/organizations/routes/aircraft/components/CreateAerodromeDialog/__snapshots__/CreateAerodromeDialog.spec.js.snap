// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes organizations routes aircraft components CreateAerodromeDialog renders correctly 1`] = `
"<div role=\\"presentation\\" class=\\"MuiDialog-root\\" data-cy=\\"aerodrome-create-dialog\\" style=\\"position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;\\">
  <div class=\\"MuiBackdrop-root\\" aria-hidden=\\"true\\" style=\\"opacity: 1; -webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\"></div>
  <div tabindex=\\"0\\" data-test=\\"sentinelStart\\"></div>
  <div class=\\"MuiDialog-container MuiDialog-scrollPaper\\" style=\\"opacity: 1; -webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\" role=\\"document\\" tabindex=\\"-1\\">
    <div class=\\"MuiPaper-root MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper CreateAerodromeDialog-root-1 MuiDialog-paperWidthSm MuiPaper-rounded\\" role=\\"dialog\\">
      <div class=\\"MuiDialogTitle-root\\">
        <h2 class=\\"MuiTypography-root MuiTypography-h6\\">Flugplatz erstellen</h2>
      </div>
      <form>
        <div class=\\"MuiDialogContent-root CreateAerodromeDialog-root-1\\">
          <div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth\\" data-cy=\\"identification-field\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink Mui-focused Mui-focused Mui-required Mui-required\\" data-shrink=\\"true\\">ICAO-Code<span class=\\"MuiFormLabel-asterisk MuiInputLabel-asterisk\\"> *</span></label>
            <div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth Mui-focused Mui-focused MuiInputBase-formControl MuiInput-formControl\\"><input aria-invalid=\\"false\\" class=\\"MuiInputBase-input MuiInput-input\\" required=\\"\\" type=\\"text\\" value=\\"\\"></div>
          </div>
          <div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth\\" data-cy=\\"name-field\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required\\" data-shrink=\\"false\\">Name<span class=\\"MuiFormLabel-asterisk MuiInputLabel-asterisk\\"> *</span></label>
            <div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl\\"><input aria-invalid=\\"false\\" class=\\"MuiInputBase-input MuiInput-input\\" required=\\"\\" type=\\"text\\" value=\\"\\"></div>
          </div>
          <div class=\\"IntegrationReactSelect-root-134\\">
            <div class=\\" css-2b097c-container\\">
              <div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink Mui-required Mui-required\\" data-shrink=\\"true\\">Zeitzone<span class=\\"MuiFormLabel-asterisk MuiInputLabel-asterisk\\"> *</span></label>
                <div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl\\">
                  <div aria-invalid=\\"false\\" class=\\"MuiInputBase-input MuiInput-input IntegrationReactSelect-input-135\\" required=\\"\\" type=\\"text\\">
                    <div class=\\"IntegrationReactSelect-valueContainer-136\\">
                      <p class=\\"MuiTypography-root IntegrationReactSelect-placeholder-140 MuiTypography-body1 MuiTypography-colorTextSecondary\\"></p>
                      <div class=\\"css-13fspk8\\">
                        <div class=\\"\\" style=\\"display: inline-block;\\"><input autocapitalize=\\"none\\" autocomplete=\\"off\\" autocorrect=\\"off\\" id=\\"react-select-2-input\\" spellcheck=\\"false\\" tabindex=\\"0\\" type=\\"text\\" aria-autocomplete=\\"list\\" style=\\"box-sizing: content-box; width: 2px; border: 0px; font-size: inherit; opacity: 1; outline: 0; padding: 0px;\\" value=\\"\\">
                          <div style=\\"position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-size: inherit; letter-spacing: normal; text-transform: none;\\"></div>
                        </div>
                      </div>
                    </div>
                    <div class=\\" css-1wy0on6\\"><span class=\\" css-1okebmr-indicatorSeparator\\"></span>
                      <div aria-hidden=\\"true\\" class=\\" css-1lwklm9-indicatorContainer\\"><svg height=\\"20\\" width=\\"20\\" viewBox=\\"0 0 20 20\\" aria-hidden=\\"true\\" focusable=\\"false\\" class=\\"css-19bqh2r\\">
                          <path d=\\"M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z\\"></path>
                        </svg></div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class=\\"MuiDialogActions-root MuiDialogActions-spacing\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">Abbrechen</span><span class=\\"MuiTouchRipple-root\\"></span></button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary\\" tabindex=\\"0\\" type=\\"submit\\" data-cy=\\"create-button\\"><span class=\\"MuiButton-label\\">Flugplatz erstellen</span><span class=\\"MuiTouchRipple-root\\"></span></button></div>
      </form>
    </div>
  </div>
  <div tabindex=\\"0\\" data-test=\\"sentinelEnd\\"></div>
</div>"
`;

exports[`routes organizations routes aircraft components CreateAerodromeDialog renders readonly if submitting 1`] = `
"<div role=\\"presentation\\" class=\\"MuiDialog-root\\" data-cy=\\"aerodrome-create-dialog\\" style=\\"position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;\\">
  <div class=\\"MuiBackdrop-root\\" aria-hidden=\\"true\\" style=\\"opacity: 1; -webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\"></div>
  <div tabindex=\\"0\\" data-test=\\"sentinelStart\\"></div>
  <div class=\\"MuiDialog-container MuiDialog-scrollPaper\\" style=\\"opacity: 1; -webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\" role=\\"document\\" tabindex=\\"-1\\">
    <div class=\\"MuiPaper-root MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper CreateAerodromeDialog-root-1 MuiDialog-paperWidthSm MuiPaper-rounded\\" role=\\"dialog\\">
      <div class=\\"MuiDialogTitle-root\\">
        <h2 class=\\"MuiTypography-root MuiTypography-h6\\">Flugplatz erstellen</h2>
      </div>
      <form>
        <div class=\\"MuiDialogContent-root CreateAerodromeDialog-root-1\\">
          <div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth\\" data-cy=\\"identification-field\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-disabled Mui-disabled Mui-required Mui-required\\" data-shrink=\\"false\\">ICAO-Code<span class=\\"MuiFormLabel-asterisk MuiInputLabel-asterisk\\"> *</span></label>
            <div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline Mui-disabled Mui-disabled MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl\\"><input aria-invalid=\\"false\\" class=\\"MuiInputBase-input MuiInput-input Mui-disabled Mui-disabled\\" disabled=\\"\\" required=\\"\\" type=\\"text\\" value=\\"\\"></div>
          </div>
          <div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth\\" data-cy=\\"name-field\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-disabled Mui-disabled Mui-required Mui-required\\" data-shrink=\\"false\\">Name<span class=\\"MuiFormLabel-asterisk MuiInputLabel-asterisk\\"> *</span></label>
            <div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline Mui-disabled Mui-disabled MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl\\"><input aria-invalid=\\"false\\" class=\\"MuiInputBase-input MuiInput-input Mui-disabled Mui-disabled\\" disabled=\\"\\" required=\\"\\" type=\\"text\\" value=\\"\\"></div>
          </div>
          <div class=\\"IntegrationReactSelect-root-134\\">
            <div class=\\" css-14jk2my-container\\">
              <div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink Mui-disabled Mui-disabled Mui-required Mui-required\\" data-shrink=\\"true\\">Zeitzone<span class=\\"MuiFormLabel-asterisk MuiInputLabel-asterisk\\"> *</span></label>
                <div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline Mui-disabled Mui-disabled MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl\\">
                  <div aria-invalid=\\"false\\" class=\\"MuiInputBase-input MuiInput-input IntegrationReactSelect-input-135 Mui-disabled Mui-disabled\\" disabled=\\"\\" required=\\"\\" type=\\"text\\">
                    <div class=\\"IntegrationReactSelect-valueContainer-136\\">
                      <p class=\\"MuiTypography-root IntegrationReactSelect-placeholder-140 MuiTypography-body1 MuiTypography-colorTextSecondary\\"></p>
                      <div class=\\"css-13fspk8\\">
                        <div class=\\"\\" style=\\"display: inline-block;\\"><input disabled=\\"\\" autocapitalize=\\"none\\" autocomplete=\\"off\\" autocorrect=\\"off\\" id=\\"react-select-3-input\\" spellcheck=\\"false\\" tabindex=\\"0\\" type=\\"text\\" aria-autocomplete=\\"list\\" style=\\"box-sizing: content-box; width: 2px; border: 0px; font-size: inherit; opacity: 1; outline: 0; padding: 0px;\\" value=\\"\\">
                          <div style=\\"position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-size: inherit; letter-spacing: normal; text-transform: none;\\"></div>
                        </div>
                      </div>
                    </div>
                    <div class=\\" css-1wy0on6\\"><span class=\\" css-109onse-indicatorSeparator\\"></span>
                      <div aria-hidden=\\"true\\" class=\\" css-1lwklm9-indicatorContainer\\"><svg height=\\"20\\" width=\\"20\\" viewBox=\\"0 0 20 20\\" aria-hidden=\\"true\\" focusable=\\"false\\" class=\\"css-19bqh2r\\">
                          <path d=\\"M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z\\"></path>
                        </svg></div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class=\\"MuiDialogActions-root MuiDialogActions-spacing\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text Mui-disabled Mui-disabled\\" tabindex=\\"-1\\" type=\\"button\\" disabled=\\"\\"><span class=\\"MuiButton-label\\">Abbrechen</span></button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled\\" tabindex=\\"-1\\" type=\\"submit\\" disabled=\\"\\" data-cy=\\"create-button\\"><span class=\\"MuiButton-label\\"><div class=\\"MuiCircularProgress-root CreateAerodromeDialog-loadingIndicator-2 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate\\" style=\\"width: 16px; height: 16px;\\" role=\\"progressbar\\"><svg class=\\"MuiCircularProgress-svg\\" viewBox=\\"22 22 44 44\\"><circle class=\\"MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate\\" cx=\\"44\\" cy=\\"44\\" r=\\"20.2\\" fill=\\"none\\" stroke-width=\\"3.6\\"></circle></svg></div>Flugplatz erstellen</span></button></div>
      </form>
    </div>
  </div>
  <div tabindex=\\"0\\" data-test=\\"sentinelEnd\\"></div>
</div>"
`;
